src/main/common/kalman.c

```diff
@@ -49,22 +49,31 @@ void kalman_init(void)
     init_kalman(&kalmanFilterStateRate[Z],  gyroConfig()->imuf_yaw_q);
 }
 
-void update_kalman_covariance(float gyroRateData, int axis)
+void update_kalman_covariance(kalman_t *kalmanState, float rate)
 {
-     kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex] = gyroRateData;
-     kalmanFilterStateRate[axis].axisSumMean +=  kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex];
-     kalmanFilterStateRate[axis].axisSumVar =  kalmanFilterStateRate[axis].axisSumVar + ( kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex] *  kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex]);
-     kalmanFilterStateRate[axis].windex++;
-    if ( kalmanFilterStateRate[axis].windex >= kalmanFilterStateRate[axis].w)
-    {
-         kalmanFilterStateRate[axis].windex = 0;
-    }
-     kalmanFilterStateRate[axis].axisSumMean -=  kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex];
-     kalmanFilterStateRate[axis].axisSumVar =  kalmanFilterStateRate[axis].axisSumVar - ( kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex] *  kalmanFilterStateRate[axis].axisWindow[ kalmanFilterStateRate[axis].windex]);
-     kalmanFilterStateRate[axis].axisMean =  kalmanFilterStateRate[axis].axisSumMean *  kalmanFilterStateRate[axis].inverseN;
-     kalmanFilterStateRate[axis].axisVar =  fabsf(kalmanFilterStateRate[axis].axisSumVar *  kalmanFilterStateRate[axis].inverseN - ( kalmanFilterStateRate[axis].axisMean *  kalmanFilterStateRate[axis].axisMean));
-
-    kalmanFilterStateRate[axis].r = sqrtf(kalmanFilterStateRate[axis].axisVar) * VARIANCE_SCALE;
+  kalmanState->axisWindow[kalmanState->windex] = rate;
+
+  kalmanState->axisSumMean += kalmanState->axisWindow[kalmanState->windex];
+  float varianceElement = kalmanState->axisWindow[kalmanState->windex] - kalmanState->axisMean;
```

Comment:
@DzikuVx , sir, why doesn't the initial value of `axisMean` need to be calculated?  is it assumed to initialize at 0? i see it without initial value calcs in other code too.